### ISSUE TITLE
create_graph in DistributedDataParallel caused memory leak

### ISSUE THREAD

**Issue by xmty777** [pytorch/pytorch#48531]
Hi@albanD  , I have a similar problem which was mentioned in [#47562](https://github.com/pytorch/pytorch/issues/47562) when I required second derivatives in DDP .
When I use `loss.backward(create_graph=True)`  in  DistributedDataParallel . It causes memory leak . (It is ok when I don't use DistributedDataParallel)
Then I find [#3818](https://github.com/pytorch/pytorch/issues/3818) and change `backward` to `autograd.grad` .
Through memory leak disappeared , it seems that `torch.autograd.grad()` can't call the autograd hook.
 
So,how should I use `create_graph` in DDP ?

_Originally posted by @xmty777 in https://github.com/pytorch/pytorch/issues/47562#issuecomment-734158759_

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd

**Comment by albanD** [comment]
Hi,

Please avoid double posting you comments...

The create_graph=True leak with backward is documented and expected. And DPP does not support it AFAIK.
